core/src/main/java/jeeves/server/dispatchers/ServiceManager.java

```diff
@@ -571,6 +572,33 @@ public ServiceContext createServiceContext(String name, int defaultUserId){
         return context;
     }
 
+    /**
+     * Provide best-guess language code (covering special case for 'fra' and 'slk' locales).
+     * <p>
+     * Logic matches LanguageUtils from module services.
+     * </p>
+     *
+     * @param locale Locale used to obtain {@link Locale#getISO3Language()} three-letter abbreviation
+     * @return language code, or "?" for unknown
+     */
+    private static String getIso3langCode(Locale locale){
+        if( locale == null ){
+            return "?";
```

Comment:
Not sure if would be better to return the default language? See https://github.com/geonetwork/core-geonetwork/blob/main/core/src/main/java/org/fao/geonet/web/DefaultLanguage.java

If the proposal seem fine, would be good also to update https://github.com/geonetwork/core-geonetwork/blob/232ed6b639a518bbb4e6f84e277262365309cc6e/core/src/main/java/jeeves/server/dispatchers/ServiceManager.java#L465

It would avoid other checks for the value `?`

Reply:
I would prefer this approach @josegar74, when doing the service factor refactor .... it was a refactor and I did not wish to change any functionality if at all possible.

Reply:
Sounds a good approach, thanks.